Doc/library/typing.rst

```diff
@@ -2341,11 +2341,22 @@ Functions and decorators
                case _ as unreachable:
                    assert_never(unreachable)
 
+   Here, the type annotations allow the type checker to infer that the
+   last case can never execute, because ``arg`` is either
+   an :class:`int` or a :class:`str`, and both options are covered by
+   earlier cases.
    If a type checker finds that a call to ``assert_never()`` is
-   reachable, it will emit an error.
+   reachable, it will emit an error. For example, if the type annotation
+   for ``arg`` was instead ``int | str | float``, the type checker would
+   emit an error pointing out that ``unreachable`` is of type :class:`float`.
```

Comment:
```suggestion
   emit an error pointing out that ``unreachable`` is of type :class:`float`.
   For a call to ``assert_never`` to succeed, the inferred type of
   ``unreachable`` must be the bottom type, :data:`Never`, and nothing else.
```

Reply:
Thanks, applied both of your suggestions.